console/views.py

```diff
@@ -65,3 +71,53 @@ def decrypt():
 
     else:
         return render_template('decrypt.html')
+
+
+def get_all_responses():
+    data = SurveyResponse.query.all()
+    return data
+
+
+def get_store_responses(tx_id, ru_ref, survey_id, datetime_earliest, datetime_latest):
+    responses = get_all_responses()
+    store_data = []
+    for response in responses:
+        store_data.append(response.data)
+
+    filtered_data = []
+    for data in store_data:
+        if (tx_id == '') or (data['tx_id'] == tx_id):
+            if (ru_ref == '') or (data['metadata']['ru_ref'] == ru_ref):
+                if (survey_id == '') or (data['survey_id'] == survey_id):
+                    filtered_data.append(data)
```

Comment:
I've managed to implement this for everything except the datetimes now. Looks like this could be a little bit trickier to do in database since the datetime is just a string inside JSON. I'll look at it again tomorrow.